prime-angular-frontend/src/app/modules/adjudication/shared/components/health-authority-table/health-authority-table.component.ts

```diff
@@ -41,27 +45,108 @@ export class HealthAuthorityTableComponent implements OnInit {
+    this.dataSource = new MatTableDataSource<HealthAuthorityListItem | HealthAuthoritySite>([]);
     this.healthAuthorityCode = this.activatedRoute.snapshot.params.haid;
+    // No HA headers should be expanded initially
+    this.idOfExpandedHA = -1;
   }
 
   public onRoute(routePath: string | (string | number)[]) {
     this.route.emit(routePath);
   }
 
+  // TODO: Eliminate duplication as this method copied from SiteRegistrationTableComponent
+  public displayStatus(status: SiteStatusType) {
+    switch (status) {
+      case SiteStatusType.EDITABLE:
+        return "Editable";
+      case SiteStatusType.IN_REVIEW:
+        return "In Review";
+      case SiteStatusType.LOCKED:
+        return "Locked";
+      default:
+        return "Editable";
+    }
+  }
+
+  // TODO: Eliminate duplication as this method very similar to that from SiteRegistrationTableComponent
+  public remoteUsers(siteRegistration: HealthAuthoritySite): number | 'Yes' | 'No' {
+    const count = siteRegistration.remoteUsers?.length;
+
+    return (!this.activatedRoute.snapshot.params.sid)
+      ? (count) ? 'Yes' : 'No'
+      : count;
+  }
+
+  public isHealthAuthorityObject(rowData): boolean {
+    return rowData.hasUnderReviewUsers !== undefined;
+  }
+
+  public isGroup(index: number, rowData: HealthAuthorityListItem | HealthAuthoritySite): boolean {
+    return HealthAuthorityTableComponent.isHA(rowData);
+  }
+
+  public expandHeader(item: HealthAuthorityListItem) {
+    this.idOfExpandedHA =
+      this.idOfExpandedHA !== item.id ? item.id : -1;
+  }
+
   public ngOnInit(): void {
     this.healthAuthorityResource.getHealthAuthorities()
-      .subscribe((healthAuthorities: HealthAuthorityList[]) => {
-        this.flaggedHealthAuthorities = healthAuthorities.reduce((fhas: number[], ha: HealthAuthorityList) =>
-          [...fhas, ...ArrayUtils.insertIf(ha.hasUnderReviewUsers, ha.id)], []
-        );
-
-        this.dataSource.data = (this.healthAuthorityCode)
-          ? healthAuthorities.filter(ha => ha.id === +this.healthAuthorityCode)
-          : healthAuthorities.sort((a, b) => a.id - b.id);
+      .pipe(
+        exhaustMap((healthAuthorities: HealthAuthorityListItem[]) => {
+          this.flaggedHealthAuthorities = healthAuthorities.reduce((fhas: number[], ha: HealthAuthorityListItem) =>
+            [...fhas, ...ArrayUtils.insertIf(ha.hasUnderReviewUsers, ha.id)], []
+          );
+
+          this.dataSource.data = (this.healthAuthorityCode)
+            ? healthAuthorities.filter(ha => ha.id === +this.healthAuthorityCode)
+            : healthAuthorities.sort(this.sortData);
+          return from(healthAuthorities);
+        }),
+        // As each healthAuthority is emitted, get the sites registered with that healthAuthority
+        mergeMap((healthAuthority: HealthAuthorityListItem) => this.healthAuthorityResource.getHealthAuthoritySites(healthAuthority.id))
+      )
+      .subscribe((sites: HealthAuthoritySite[]) => {
```

Comment:
Also I think this was the cause of the "some time" sorting error. Since you are calling 7 async calls that when they return add the sites and sort them into the datasource.